setup/helpers/perf_helper.rb

```diff
@@ -343,19 +249,33 @@ def perf_install_foss
       if package_build_version
         Beaker::Log.notify("Installing OSS Puppet Server version '#{package_build_version}'")
         install_puppetlabs_dev_repo master, 'puppetserver', package_build_version,
-                                    repo_config_dir, @install_opts
-      else
+                                    repo_config_dir, install_opts
+        else
         abort("Environment variable PACKAGE_BUILD_VERSION required for package installs!")
       end
+    end
 
-      on master, 'puppet resource service puppetserver ensure=running enable=true'
+    step "Upgrade nss to version that is hopefully compatible with jdk version puppetserver will use." do
+      nss_package=nil
+      variant, _, _, _ = master['platform'].to_array
+      case variant
+        when /^(debian|ubuntu)$/
+          nss_package_name="libnss3"
+        when /^(redhat|el|centos)$/
+          nss_package_name="nss"
+      end
+      if nss_package_name
+        master.upgrade_package(nss_package_name)
+      else
+        Beaker::Log.warn("Don't know what nss package to use for #{variant} so not installing one")
+      end
     end
 
-    step "Sign agent certificates" do
-      # Failed agent run will request a certificate
-      on agents, 'puppet agent -t', {:accept_all_exit_codes => true}
-      on master, 'puppet cert sign --all'
```

Comment:
where did the cert signing go?  I don't see anything signing certs to make up for this line being removed.

Reply:
It is part of the normal setup. I had to call it explicitly before because I couldn't use the default setup method, but now that we have access to internal resources it uses the default setup which takes care of it.